tao_compiler/mlir/disc/transforms/disc_shape_optimization_utils.cc

```diff
@@ -385,57 +400,108 @@ LogicalResult SymbolicDimMgr::updateProductEqualityMap() {
       LLVM_DEBUG(llvm::dbgs() << "Pair: x = " << x << "\ny = " << y << "\n");
     }
   }
+  productEqualityMapUpdated_ = true;
```

Comment:
Just curiouse why not using a union-find set to track equal products?

Reply:
we need to do some simplification.

`a0 * b0 * c0 * d0 = a0 * b1 * c1 * d1` no need to be stored if we know `a0 * b0` == 'a0 * b1' and `c0 * d0 == c1 * d1`.
or
`a0 * b0 * c0 == a0 * d1 * e1` -> `b0 * c0 == d1 * e1` if we know `a0 != 0`.

Thus, we need to update the product equality map during new insertion.